src/rnn.jl

```diff
@@ -159,32 +159,37 @@ function cudnnGetFilterNdDescriptor(wDesc::FD; nbDimsRequested = 8)
     end
 end
 
+
+# call gpu everytime to reflect device changes
+gethandle() = gpu() >= 0 ? cudnnhandle() : nothing
+
+
 """
 
-    rnnparam{T}(r::RNN, w::KnetArray{T}, layer, id, param)
+        rnnparam{T}(r::RNN, w::KnetArray{T}, layer, id, param)
 
-Return a single weight matrix or bias vector as a slice of w.
+    Return a single weight matrix or bias vector as a slice of w.
 
-Valid `layer` values:
-* For unidirectional RNNs 1:numLayers
-* For bidirectional RNNs 1:2*numLayers, forw and back layers alternate.
+    Valid `layer` values:
+    * For unidirectional RNNs 1:numLayers
+    * For bidirectional RNNs 1:2*numLayers, forw and back layers alternate.
 
-Valid `id` values:
-* For RELU and TANH RNNs, input = 1, hidden = 2.
-* For GRU reset = 1,4; update = 2,5; newmem = 3,6; 1:3 for input, 4:6 for hidden
-* For LSTM inputgate = 1,5; forget = 2,6; newmem = 3,7; output = 4,8; 1:4 for input, 5:8 for hidden
+    Valid `id` values:
+    * For RELU and TANH RNNs, input = 1, hidden = 2.
+    * For GRU reset = 1,4; update = 2,5; newmem = 3,6; 1:3 for input, 4:6 for hidden
+    * For LSTM inputgate = 1,5; forget = 2,6; newmem = 3,7; output = 4,8; 1:4 for input, 5:8 for hidden
 
-Valid `param` values:
-* Return the weight matrix (transposed!) if `param==1`.
-* Return the bias vector if `param==2`.
+    Valid `param` values:
+    * Return the weight matrix (transposed!) if `param==1`.
+    * Return the bias vector if `param==2`.
 
-The effect of skipInput: Let I=1 for RELU/TANH, 1:3 for GRU, 1:4 for LSTM
-* For skipInput=false (default), rnnparam(r,w,1,I,1) is a (inputSize,hiddenSize) matrix.
-* For skipInput=true, rnnparam(r,w,1,I,1) is `nothing`.
-* For bidirectional, the same applies to rnnparam(r,w,2,I,1): the first back layer.
+    The effect of skipInput: Let I=1 for RELU/TANH, 1:3 for GRU, 1:4 for LSTM
+    * For skipInput=false (default), rnnparam(r,w,1,I,1) is a (inputSize,hiddenSize) matrix.
+    * For skipInput=true, rnnparam(r,w,1,I,1) is `nothing`.
+    * For bidirectional, the same applies to rnnparam(r,w,2,I,1): the first back layer.
 
-"""
-function rnnparam(r::RNN, w, layer::Integer, id::Integer, par::Integer; handle=cudnnhandle())
+    """
+function rnnparam(r::RNN, w, layer::Integer, id::Integer, par::Integer; handle=gethandle(), useview=false)
```

Comment:
Do we need useview to be an option?  We can just pick the more efficient implementation.